editoast/src/views/v2/path/pathfinding.rs

```diff
@@ -95,117 +94,201 @@ pub async fn post(
+        futures.push(Box::pin(request.fetch(core.as_ref())));
     }
+    let computed_paths: Vec<_> = futures::future::join_all(futures)
+        .await
+        .into_iter()
+        .collect::<Result<_>>()?;
 
-    // 2) Compute path from core
-    let pathfinding_request = PathfindingRequest {
-        infra: infra.id,
-        expected_version: infra.version.clone(),
-        path_items: track_offsets,
-        rolling_stock_loading_gauge: path_input.rolling_stock_loading_gauge,
-        rolling_stock_is_thermal: path_input.rolling_stock_is_thermal,
-        rolling_stock_supported_electrifications: path_input
-            .rolling_stock_supported_electrifications
-            .clone(),
-        rolling_stock_supported_signaling_systems: path_input
-            .rolling_stock_supported_signaling_systems
-            .clone(),
-    };
-    let pathfinding_result = pathfinding_request.fetch(core.as_ref()).await?;
+    for (index, computed_path) in computed_paths.into_iter().enumerate() {
+        let path_index = pathfinding_requests_index[index];
+        to_cache.push((&hashes[path_index], computed_path.clone()));
+        pathfinding_results[path_index] = Some(computed_path);
+    }
 
     // 3) Put in cache
-    redis_conn
-        .json_set_ex(&hash, &pathfinding_result, CACHE_PATH_EXPIRATION)
-        .await?;
+    redis_conn.json_set_bulk(&to_cache).await?;
 
-    Ok(pathfinding_result)
+    Ok(pathfinding_results.into_iter().flatten().collect())
 }
 
-/// Compute a path given a batch of trainschedule and an infrastructure.
-///
-/// ## Important
-///
-/// If this function was called with the same train schedule, the result will be cached.
-/// If you call this function multiple times with the same train schedule but with another infra, then you must provide a fresh `cache`.
+/// Compute a path given a train schedule and an infrastructure.
 pub async fn pathfinding_from_train(
     conn: &mut DbConnection,
     redis: &mut RedisConnection,
     core: Arc<CoreClient>,
     infra: &Infra,
     train_schedule: TrainSchedule,
-    proxy: Arc<TrainScheduleProxy>,
 ) -> Result<PathfindingResult> {
-    if let Some(res) = proxy.get_pathfinding_result(train_schedule.id) {
-        return Ok(res);
+    let rolling_stocks =
+        RollingStockModel::retrieve(conn, train_schedule.rolling_stock_name.clone())
+            .await?
+            .into_iter()
+            .map(|rs| (rs.name.clone(), rs))
+            .collect();
+
+    Ok(
+        pathfinding_from_train_batch(conn, redis, core, infra, &[train_schedule], &rolling_stocks)
+            .await?
+            .pop()
+            .unwrap(),
+    )
+}
+
+/// Compute a path given a batch of trainschedule and an infrastructure.
+pub async fn pathfinding_from_train_batch(
+    conn: &mut DbConnection,
+    redis: &mut RedisConnection,
+    core: Arc<CoreClient>,
+    infra: &Infra,
+    train_schedules: &[TrainSchedule],
+    rolling_stocks: &HashMap<String, RollingStockModel>,
```

Comment:
nit: it looks cumbersome to handle a separate mapping of rolling stocks, maybe we could introduce some sort of "FullTrainSchedule" struct, which would hold to it's rolling stock ?

Reply:
Good idea but I think I'll do it later